airflow/datasets/__init__.py

```diff
@@ -314,6 +361,38 @@ def as_expression(self) -> Any:
         """
         return {"alias": self.name}
 
+    def iter_dag_dependencies(self, *, source: str = "", target: str = "") -> Iterator[DagDependency]:
+        """
+        Iterate a dataset alias and its resolved datasets  as dag dependency.
+
+        :meta private:
+        """
+        if self.objects:
+            for obj in self.objects:
+                dataset = cast(Dataset, obj)
```

Comment:
I don’t think this is right? We can have nesting conditions, in those cases `obj` would not be a Dataset.

Reply:
No nested condition would happen in `_DatasetAliasCondition` (at least in the current design) . This condition will only expand a `DatasetAlias` to `DatasetAny( resolved datasets )`